README.md

```diff
@@ -11,6 +11,18 @@ Selection of useful reusable components
 
 ---
 
+# Collection utils
+
+```scala
+import CollectionImplicits._
+
+val iter: Iterator[_] = ???
+val arr: Array[_] = ???
+
+// copy 42 items from the `iter` to the `arr` with array offset 7
+iter.fetch(arr, 7, 42) // returns a number of actually copied items
```

Comment:
To me fetch just doesn't sound right. I would expect the iterator to fetch me something for example return the array. 

Why not use `copyWithCount` ?

Reply:
The original method is called `copyToArray`.  So what about `fetchToArray` then?

I'm not keen on using `WithCount` in this context as it sounds like you "copy items with count", while in fact you only copy items along, and then return they count. So personally I'd avoid mixing what a method does (a side-effect) with what it returns as a function in one name.

Reply:
`fetchToArray` sounds good.